src/store/calendars.js

```diff
@@ -123,12 +172,379 @@ const getters = {
+			return list
+		}, calendar.events)
+	},
+
+	/**
+	 * Add an event to an calendar and overwrite if duplicate uid
+	 *
+	 * @param {Object} state the store data
+	 * @param {Event} event the event to add
+	 */
+	addEventToCalendar(state, event) {
+		let calendar = state.calendars.find(search => search.id === event.calendar.id)
+		Vue.set(calendar.events, event.uid, event)
+	},
+
+	/**
+	 * Delete an event in a specified calendar
+	 *
+	 * @param {Object} state the store data
+	 * @param {Event} event the event to delete
+	 */
+	deleteEventFromCalendar(state, event) {
+		let calendar = state.calendars.find(search => search.id === event.calendar.id)
+		Vue.delete(calendar, event.uid)
+	},
 
-const actions = {}
+	/**
+	 * Share calendar with a user or group
+	 *
+	 * @param {Object} state the store data
+	 * @param {Object} data destructuring object
+	 * @param {Object} data.calendar the calendar
+	 * @param {string} data.sharee the sharee
+	 * @param {string} data.id id
+	 * @param {Boolean} data.group group
+	 */
+	shareCalendar(state, { calendar, sharee, id, group }) {
+		calendar = state.calendars.find(search => search.id === calendar.id)
+		let newSharee = {
+			displayname: sharee,
+			id,
+			writeable: false,
+			group
+		}
+		calendar.shares.push(newSharee)
+	},
+
+	/**
+	 * Remove Sharee from calendar shares list
+	 *
+	 * @param {Object} state the store data
+	 * @param {Object} sharee the sharee
+	 */
+	removeSharee(state, sharee) {
+		let calendar = state.calendars.find(search => {
+			for (let i in search.shares) {
+				if (search.shares[i] === sharee) {
+					return true
+				}
+			}
+		})
+		calendar.shares.splice(calendar.shares.indexOf(sharee), 1)
+	},
+
+	/**
+	 * Toggle sharee's writable permission
+	 *
+	 * @param {Object} state the store data
+	 * @param {Object} sharee the sharee
+	 */
+	updateShareeWritable(state, sharee) {
+		let calendar = state.calendars.find(search => {
+			for (let i in search.shares) {
+				if (search.shares[i] === sharee) {
+					return true
+				}
+			}
+		})
+		sharee = calendar.shares.find(search => search === sharee)
+		sharee.writeable = !sharee.writeable
+	}
+}
+
+const actions = {
+	/**
+	 * Retrieve and commit calendars
+	 *
+	 * @param {Object} context the store mutations
+	 * @returns {Promise<Array>} the calendars
+	 */
+	async getCalendars(context) {
+		let calendars = await client.calendarHomes[0].findAllCalendars()
+			.then(calendars => {
+				return calendars.map(calendar => {
+					return mapDavCollectionToCalendar(calendar)
+				})
+			})
+
+		calendars.forEach(calendar => {
+			context.commit('addCalendar', calendar)
+		})
+
+		return calendars
+	},
+
+	/**
+	 * Append a new calendar to array of existing calendars
+	 *
+	 * @param {Object} context the store mutations
+	 * @param {Object} calendar The calendar to append
+	 * @returns {Promise}
+	 */
+	async appendCalendar(context, calendar) {
+		return client.calendarHomes[0].createCalendarCollection(calendar.displayName)
```

Comment:
@skjnldsv @georgehrke Is there a way to specify the calendar color on creation? Currently no calendar color is given which results in a transparent calendar icon. You can change the color later, though.

Reply:
@raimund-schluessler there should be a way to send the proper keys on create I guess, but you need to see with @georgehrke for that ;)

Reply:
Color is the second parameter :)
https://github.com/nextcloud/cdav-library/blob/master/src/models/calendarHome.js#L108

I will also add a third parameter to define supported calendar component sets, so that you can create "pure" VTodo calendars.

Reply:
> I will also add a third parameter to define supported calendar component sets, so that you can create "pure" VTodo calendars.

Perfect.